### PR TITLE
grammar, usage, punctuation edits, Part 1, sections 3.1 - 4.6

### PR DESCRIPTION
**Pull request by jmbothe**

### CODE REVIEW

**Comment by iliakan** [comment]
@jmbothe What you think about this PR: https://github.com/iliakan/javascript-tutorial-en/pull/139/files, is it a correct change?

File: 1-js/03-code-quality/03-comments/article.md [hunk 1]
```diff
@@ -145,6 +145,6 @@ Any subtle features of the code? Where they are used?
 
 ## Summary
 
-One of signs of a good developer is his comments. Good comments allow to maintain the code well, return to it after a long delay and use features more effectively.
+One sign of a good developer is his comments. Good comments allowus  to maintain the code well, return to it after a long delay and use features more effectively.
```
Comment (iliakan) [inline 1-js/03-code-quality/03-comments/article.md]: a space lost?

File: 1-js/03-code-quality/04-ninja-code/article.md [hunk 2]
```diff
@@ -66,17 +66,17 @@ The great image has no form.
 
 While choosing a name try to use the most abstract word. Like `obj`, `data`, `value`, `item`, `elem` and so on.
 
-- **The ideal name for a variable is `data`.** Use it everywhere where you can. Indeed, every variable holds *data*, right?
+- **The ideal name for a variable is `data`.** Use it everywhere you can. Indeed, every variable holds *data*, right?
```
Comment (iliakan) [inline 1-js/03-code-quality/04-ninja-code/article.md]: "Use it everywhere you can" - does that sound good?

File: 1-js/03-code-quality/03-comments/article.md (hunk 1 above)
Comment (iliakan) [inline 1-js/03-code-quality/03-comments/article.md]: guess, a typo fix please.

File: 1-js/03-code-quality/04-ninja-code/article.md [hunk 3]
```diff
@@ -223,6 +223,6 @@ Imagine, another developer wants only to check the email, and not output any mes
 
 All "pieces of advice" above are from the real code... Sometimes, written by experienced developers. Maybe even more experienced than you are ;)
 
-- Follow some of them -- and your code will become full of surprises.
-- Follow many of them -- and your code will become truly yours, no one would want to change it.
-- Follow all -- and your code will become a valuable lesson for young developers looking for enlightment.
+- Follow some of them, and your code will become full of surprises.
```
Comment (iliakan) [inline 1-js/03-code-quality/04-ninja-code/article.md]: Are you sure that a dash is wrong here? The phrases look better with it. Are the rules  really so strict that there must be no dash, or maybe a kind of "author punctuation" is possible?

File: 1-js/03-code-quality/04-ninja-code/article.md (hunk 2 above)
Comment (jmbothe) [inline 1-js/03-code-quality/04-ninja-code/article.md]: @iliakan This change is grammatically correct. The extra "where" is redundant. And to answer "does it sound good here": yes, this sounds like natural English. It fits your conversational style well, I think :)

File: 1-js/03-code-quality/04-ninja-code/article.md (hunk 3 above)
Comment (jmbothe) [inline 1-js/03-code-quality/04-ninja-code/article.md]: @iliakan This change is correct punctuation, but as far as whether the sentence look better with dashes: that is a stylistic choice for you to make. But considering punctuation only, the "Follow..." clauses are **dependent clauses** of the rest of the sentence. In English when a dependent clause precedes an **independent clause**, we usually place a comma between them, although the comma is optional. It is considered incorrect to separate a dependent clause and an independent clause with a dash.

File: 1-js/03-code-quality/03-comments/article.md (hunk 1 above)
Comment (jmbothe) [inline 1-js/03-code-quality/03-comments/article.md]: @iliakan Sorry! this is my mistake. I will fix it, and I will be more careful next time :)

File: 1-js/03-code-quality/04-ninja-code/article.md (hunk 3 above)
Comment (iliakan) [inline 1-js/03-code-quality/04-ninja-code/article.md]: Okay, I'd just like to introduce a pause there. Using comma is all right?

File: 1-js/03-code-quality/04-ninja-code/article.md (hunk 2 above)
Comment (iliakan) [inline 1-js/03-code-quality/04-ninja-code/article.md]: Ok great.

File: 1-js/03-code-quality/04-ninja-code/article.md (hunk 3 above)
Comment (jmbothe) [inline 1-js/03-code-quality/04-ninja-code/article.md]: Yes, a comma is good for introducing a pause here. It represent logical separation of the clauses, and a natural pause in reading/speaking rhythm.

**Pull request merged**